### PR TITLE
users can edit their own posts

### PR DESCRIPTION
**Pull request by lexidelorey**

### CODE REVIEW

**Review by Omarzintan — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/EventServlet.java [hunk 1]
```diff
@@ -37,10 +36,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String requestStartTime = request.getParameter(Constants.START_TIME_PARAM);
     String requestEndDate = request.getParameter(Constants.END_DATE_PARAM);
     String requestEndTime = request.getParameter(Constants.END_TIME_PARAM);
-    String timeZone = request.getParameter(Constants.TIME_ZONE_PARAM);
 
-    Date startDateTime = parseInputDateTime(requestStartDate, requestStartTime, timeZone);
-    Date endDateTime = parseInputDateTime(requestEndDate, requestEndTime, timeZone);
+    String isEventEditing = request.getParameter(Constants.EVENT_EDITING_PARAM);
```
Comment (rglassett) [inline src/main/java/com/google/sps/servlets/EventServlet.java:40]: Generally edits are made with the HTTP `PATCH` or `PUT` verbs. If you update the HTML `<form method="...">` attribute, you could implement the "edit" code path separately from "create", which might be cleaner.

File: src/main/java/com/google/sps/servlets/EventServlet.java [hunk 2]
```diff
@@ -57,7 +57,9 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         json, 
         requestStartTime, 
         requestEndTime, 
-        timeZone);  
+        requestStartDate, 
+        requestEndDate, 
+        isEventEditing);
```
Comment (rglassett) [inline src/main/java/com/google/sps/servlets/EventServlet.java:62]: I would avoid adding new fields to this entity unless they actually need to be stored (`isEventEditing` appears to be transient state).

File: src/main/java/com/google/sps/servlets/EventServlet.java [hunk 3]
```diff
@@ -215,13 +230,29 @@ private void createEntity(
...
+          endDate, endTimeFormatted);
 
     Entity eventEntity = new Entity(Constants.EVENT_ENTITY_PARAM);
+    if (isEventEditing.equals("yes")) {
+      long eventId = Long.parseLong(request.getParameter(Constants.EVENT_ID_PARAM));
+      Query query = new Query(Constants.EVENT_ENTITY_PARAM);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+
+      for (Entity entity : results.asIterable()) {
```
Comment (rglassett) [inline src/main/java/com/google/sps/servlets/EventServlet.java:243]: It's expensive to iterate over every record in the datastore. Instead consider adding a filter and fetching only the record you're interested in.

https://cloud.google.com/appengine/docs/standard/java/datastore/retrieving-query-results#retrieving_entities

File: src/main/java/com/google/sps/servlets/EventServlet.java [hunk 4]
```diff
@@ -235,7 +266,12 @@ private void createEntity(
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(eventEntity);
 
-    json.put("success", "true");
+    if (isEventEditing.equals("yes")) {  
+      json.put("success", "edit");
```
Comment (rglassett) [inline src/main/java/com/google/sps/servlets/EventServlet.java:271]: Mixing boolean values with strings like "edit" is a code smell that an enum type might be more appropriate.

**Review by rglassett — COMMENTED** [review]
I strongly recommend implementing a separate action and code path for updates. I think you'll see a big decrease in complexity in doing so.

**Pull request merged**